miner/miner.go

```diff
@@ -163,3 +166,100 @@ func (miner *Miner) getPending() *newPayloadResult {
+				blockIncludedFeeGauge.Update(new(big.Int).Div(payout.Value(), bigGwei).Int64())
+			} else {
+				// Possibly not an MEV block, pull in the balance change and report that
+				parent := chain.GetBlock(block.ParentHash(), block.NumberU64()-1)
+				if parent != nil {
+					oldState, oerr := chain.StateAt(parent.Root())
+					newState, nerr := chain.StateAt(block.Root())
+
+					if oerr == nil && nerr == nil {
+						balanceDiff := new(uint256.Int).Sub(newState.GetBalance(block.Coinbase()), oldState.GetBalance(block.Coinbase()))
+						diffInEther := new(big.Float).Quo(new(big.Float).SetInt(balanceDiff.ToBig()), big.NewFloat(params.Ether))
+						feeInEther := new(big.Float).Quo(new(big.Float).SetInt(payload.fullFees), big.NewFloat(params.Ether))
+
+						log.Info("Plain block detected", "block", block.NumberU64(), "reward", diffInEther, "local", feeInEther, "delta", new(big.Float).Sub(diffInEther, feeInEther))
+						blockIncludedFeeGauge.Update(new(big.Int).Div(balanceDiff.ToBig(), bigGwei).Int64())
+					} else {
+						log.Error("Missing state for reward metric", "old", oerr, "new", nerr)
+					}
+				}
+			}
+		}
+	} else if payload.full.Coinbase() != block.Coinbase() {
+		// We are not in benchmarking mode and out proposed fee recipient is
+		// different from the live reward recipient. Definitely an MEV block.
+		// Try to extract the rewards.
+		if count := len(block.Transactions()); count > 0 {
+			payout := block.Transactions()[count-1]
+			if payout.To() != nil && *payout.To() == payload.full.Coinbase() {
+				mevInEther := new(big.Float).Quo(new(big.Float).SetInt(payout.Value()), big.NewFloat(params.Ether))
+				feeInEther := new(big.Float).Quo(new(big.Float).SetInt(payload.fullFees), big.NewFloat(params.Ether))
+
+				log.Info("MEV reward received", "block", block.NumberU64(), "reward", mevInEther, "local", feeInEther, "delta", new(big.Float).Sub(mevInEther, feeInEther))
+				blockIncludedFeeGauge.Update(new(big.Int).Div(payout.Value(), bigGwei).Int64())
+			} else {
+				// Unknown MEV block, pull in the balance change and report that
+				parent := chain.GetBlock(block.ParentHash(), block.NumberU64()-1)
+				if parent != nil {
+					oldState, oerr := chain.StateAt(parent.Root())
+					newState, nerr := chain.StateAt(block.Root())
+
+					if oerr == nil && nerr == nil {
+						balanceDiff := new(uint256.Int).Sub(newState.GetBalance(payload.full.Coinbase()), oldState.GetBalance(payload.full.Coinbase()))
+						diffInEther := new(big.Float).Quo(new(big.Float).SetInt(balanceDiff.ToBig()), big.NewFloat(params.Ether))
+						feeInEther := new(big.Float).Quo(new(big.Float).SetInt(payload.fullFees), big.NewFloat(params.Ether))
+
+						log.Info("MEV hidden reward received", "block", block.NumberU64(), "reward", diffInEther, "local", feeInEther, "delta", new(big.Float).Sub(diffInEther, feeInEther))
```

Comment:
As far as I can tell, the newState.balance contains both hidden mev reward, but also the transaction fees. I think you need to subtract the tx tips from the `reward` (but not the fulll fees, in case part of the full fees are burnt?)